### ISSUE TITLE
[Core][Gamebreaking] Flying in azeroth

### ISSUE THREAD

**Issue by killablink** [Atlantiss/NetherwingBugtracker#3797]
**Description**:  If you alt+f4 after using a flight path, your mount is going to be flapping his wings 100x speed and when you reach the ground, you will be flying.
https://gfycat.com/GiganticEmbellishedArmyworm
https://gfycat.com/FlashyShamelessKakapo
**Current behaviour**:

**Expected behaviour**: Normal behavior

**Server Revision**: 2566

**Comment by Jezofine** [comment]
I made some attempts to reproduce as horde:
1) Ratchet -> Crossroads: everything's ok, after relogging I resumed my flight normally.

2) Crossroads -> Orgrimmar (2 times): after relogging I flew all the way back to Cross through textures, flight to Org then restarted automatically. I was randomly teleported around Cross, and my wyvern was flying sideways. Almost 2x flight time in total.

3) Orgrimmar -> Valormok (Azshara) (2 times): ok, tried alt-f4 early in Org and late in Aszhara.

4) Valormok -> Orgrimmar (2 times): if I alt-f4 early while still in Azshara then flight resumes correctly, but alt-f4 after or just before switching to Barrens/Orgrimmar zone results in the same behavior as 2)

None of these flights resulted in flapping wings and flying character, though.
